### PR TITLE
docs(platformAutomerge): recommend setting required status checks

### LINKED ISSUE #16964: docs: add warning to platform automerge to configure required status checks
### Describe the proposed change(s).

We should add a warning to `platformAutomerge`[^1] docs thatt this feature requires madatory status checks, otherwise PR's can get merged with failed status checks.

I know at least github will auto merge if the status checks are delayed.

[^1]: https://docs.renovatebot.com/configuration-options/#platformautomerge

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Add admonition to tell our users they _must_ set required status checks, or Renovate will automerge PRs even if all tests fail

## Context

Closes #16964

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -1904,6 +1904,15 @@ If enabled Renovate will pin Docker images by means of their SHA256 digest and n
...
 ## platformAutomerge
 
+<!-- prettier-ignore -->
+!!! warning
+    Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.
+    If you don't do this, Renovate auto-merges PRs even if all test fail!
+    On GitHub this is called "Require status checks before merging", which you can find in the "Branch protection rules" section of the settings for your repository.
+    [GitHub docs, about protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)
+    [GitHub docs, require status checks before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging)
+    If you're using another platform, search their documentation for a similar feature.
```
Comment (HonkingGoose) [inline docs/usage/configuration-options.md:1914]: I don't know how to format this text so:

- Prettier ignores the formatting
- Material for MkDocs renders the admonition correctly on the published page

Here's what the end-result should look like:

> Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.
> If you don't do this, Renovate auto-merges PRs even if all test fail!
> 
> On GitHub this is called "Require status checks before merging", which you can find in the "Branch protection rules" section of the settings for your repository.
> - [GitHub docs, about protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)
> - [GitHub docs, require status checks before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging)
>
> If you're using another platform, search their documentation for a similar feature.

If you have any tips/tricks/suggestions, please let me know. 😉

File: docs/usage/configuration-options.md [hunk 2]
```diff
@@ -1904,6 +1904,15 @@ If enabled Renovate will pin Docker images by means of their SHA256 digest and n
 
 ## platformAutomerge
 
+<!-- prettier-ignore -->
+!!! warning
+    Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.
+    If you don't do this, Renovate auto-merges PRs even if all test fail!
```
Comment (viceice) [inline docs/usage/configuration-options.md:1910]:
```suggestion
    If you don't do this, The platform auto-merges PRs even if all test fail!
```
the platform will do the auto-merge

Comment (rarkins) [inline docs/usage/configuration-options.md:1910]:
```suggestion
    If you don't do this, the platform might merge Renovate PRs even if the repository's tests haven't started, are in still in progress, or possibly even when they have failed. 
```

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -1904,6 +1904,15 @@ If enabled Renovate will pin Docker images by means of their SHA256 digest and n
 
 ## platformAutomerge
 
+<!-- prettier-ignore -->
+!!! warning
+    Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:1909]:
```suggestion
    Before you enable `platformAutomerge` you should enable your Git hosting platform's capabilities to enforce test passing before PR merge.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Pull request merged**